Client/src/Components/Graph/GraphControls/AxesControl.tsx

```diff
@@ -29,48 +31,78 @@ const buildVariableList = (datasets: IDatasetModel[]): IVariable[] => {
 }
 
 export const AxesControl = (props: IProps) => {
-  const { datasets, axes, onAxesChange } = { ...props }
+  const { datasets, axes, onAxesChange, dimensions } = { ...props }
   const classes = classStyles()
 
-  //todo unhardcode the variables
   const [showSettings, setSettingsToggle] = useState(true)
 
   const [xVariableMissing, setXVariableMissing] = useState([])
   const [yVariableMissing, setYVariableMissing] = useState([])
-  const [xUnits, setXUnits] = useState([])
-  const [yUnits, setYUnits] = useState([])
+  const [xUnits, setXUnits] = useState<IUnitModel[]>([])
+  const [yUnits, setYUnits] = useState<IUnitModel[]>([])
   const [variables, setVariables] = useState<IVariable[]>([])
 
   useEffect(() => {
     setVariables(buildVariableList(datasets))
   }, [datasets])
 
+  const getVariableDimension = (datasets: IDatasetModel[], variableName): number => {
+
+    const dictionary = {}
+
+    datasets.forEach(dataset => {
+      const foundVariable = dataset.data.variables.find(variable => variable.name == variableName)
+      if (foundVariable) {
+        const datasetIds = dictionary[foundVariable.dimensionId] || []
+        datasetIds.push(dataset.id)
+        dictionary[foundVariable.dimensionId] = datasetIds
+      }
+    })
+
+    let index = '';
+    let size = -1;
+    for (let key in dictionary) {
+      if (dictionary[key].length > size) {
```

Comment:
I dont think its possible for this condition to be false...

Reply:
@GD-Dheer this if statement checks if there exists datasets for that chosen variable on the axis but I guess setting size=-1 wasn't necessary

Reply:
The value of size changes depending on the legnth of the dictionary entry for a given dimension id